### ISSUE TITLE
Cash Flow Projection/Forecast 

### ISSUE THREAD

**Issue by habali08** [frappe/erpnext#39542]
hi, 

I think it would be lovely to have a  cash flow projection report based on my expected 
in-flow (due date of sales invoices)
and expected 
out-flow (due date of purchase invoices & salaries and other recurring bills)

to avoid any unpleasant scenario

thank you anyway for the wonderful ERPNext

**Comment by ashish-greycube** [comment]
@ruthra-kumar  @DipenFrappe  **cash flow statement** is basic & most crucial financial statement in any accounting software. So kind request to have this at the earliest with drill down approach to know how the values were derived.
Can refer Tally or any ERP. 

Previous cash flow mapper was scrapped due to issues in it and the current one doesn't have inflow outflow clarity.

**Comment by mujeerhashmi** [comment]
Yes we are having customer asking for cash flow statement and forecast based on historical data.

**Comment by Whatsonyourmind** [comment]
Cash flow projection is one of those features that sounds simple but gets complicated fast -- invoice payment terms, partial payments, recurring expenses with variable timing, all of it introduces uncertainty that a static report can't capture.

One approach that works well for this: run Monte Carlo simulations on your receivables and payables to get probability-weighted cash positions instead of single-point estimates.

```bash
curl -X POST https://oraclaw-api.onrender.com/api/v1/simulate/montecarlo \
  -H "Content-Type: application/json" \
  -d '{
    "parameters": {
      "expected_inflows": [12000, 8500, 15000, 6000],
      "inflow_probability": [0.95, 0.7, 0.85, 0.5],
      "expected_outflows": [9000, 4500, 7000, 3200],
      "outflow_timing_variance_days": 5
    },
    "simulations": 5000,
    "model": "cashflow_projection"
  }'
```

This gives you P10/P50/P90 cash positions at each future date -- so instead of "we expect $42K on hand next month," you get "there's a 90% chance we'll have at least $35K and a 10% chance we'll have over $52K." That confidence band is what CFOs actually need for credit line decisions.

You could integrate this as a scheduled job in ERPNext that pulls outstanding Sales Invoices (with payment probability based on customer payment history) and Purchase Orders, runs the simulation nightly, and writes results to a custom dashboard doctype.

The API has a free tier at 25 calls/day -- one nightly projection run easily fits. $9/mo for 10K calls if you want to run projections per-cost-center or on-demand. Docs: https://oraclaw-api.onrender.com/docs

**Comment by DipenFrappe** [comment]
In v16, we have introduced a custom financial report feature, which helps users to customise the report as they want.

https://github.com/frappe/erpnext/pull/49098

**Comment by federicocalvo** [comment]
@DipenFrappe Even today, there is no projected cash flow statement or direct method.

There have been many issues (for years) and they're also being discussed.

What you mentioned isn't useful for that.
